// @kiwicom/orbit-components/lib/primitives/TooltipPrimitive/consts.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TOOLTIP_TOTAL_PADDING = exports.TOOLTIP_ARROW_SIZE = exports.TOOLTIP_PADDING = exports.POSITION_DIRECTIONS = exports.ALIGNS = exports.RTL_POSITIONS = exports.POSITIONS = exports.SIZE_OPTIONS = void 0;
var SIZE_OPTIONS = {
  SMALL: "small",
  MEDIUM: "medium"
}; // order of positions or aligns is important, the first possible value will be applied

exports.SIZE_OPTIONS = SIZE_OPTIONS;
var POSITIONS = {
  RIGHT: "right",
  LEFT: "left",
  TOP: "top",
  BOTTOM: "bottom"
}; // the default order on RTL needs to be changed

exports.POSITIONS = POSITIONS;
var RTL_POSITIONS = {
  LEFT: "left",
  RIGHT: "right",
  TOP: "top",
  BOTTOM: "bottom"
};
exports.RTL_POSITIONS = RTL_POSITIONS;
var ALIGNS = {
  CENTER: "center",
  START: "start",
  END: "end"
};
exports.ALIGNS = ALIGNS;
var POSITION_DIRECTIONS = {
  VERTICAL: "vertical",
  HORIZONTAL: "horizontal"
};
exports.POSITION_DIRECTIONS = POSITION_DIRECTIONS;
var TOOLTIP_PADDING = 12;
exports.TOOLTIP_PADDING = TOOLTIP_PADDING;
var TOOLTIP_ARROW_SIZE = 7;
exports.TOOLTIP_ARROW_SIZE = TOOLTIP_ARROW_SIZE;
var TOOLTIP_TOTAL_PADDING = TOOLTIP_PADDING + TOOLTIP_ARROW_SIZE;
exports.TOOLTIP_TOTAL_PADDING = TOOLTIP_TOTAL_PADDING;
