// @kiwicom/orbit-components/lib/Table/TableBody/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.StyledTableBody = void 0;

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _defaultTheme = _interopRequireDefault(require("../../defaultTheme"));

var StyledTableBody = (0, _styledComponents.default)(function (_ref) {
  var children = _ref.children,
      className = _ref.className,
      dataTest = _ref.dataTest;
  return /*#__PURE__*/React.createElement("tbody", {
    className: className,
    "data-test": dataTest
  }, children);
}).withConfig({
  displayName: "TableBody__StyledTableBody",
  componentId: "sc-1uovf8k-0"
})(["width:100%;white-space:nowrap;border-bottom:1px solid ", ";"], function (_ref2) {
  var theme = _ref2.theme;
  return theme.orbit.borderColorTable;
}); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198

exports.StyledTableBody = StyledTableBody;
StyledTableBody.defaultProps = {
  theme: _defaultTheme.default
};

var TableBody = function TableBody(_ref3) {
  var children = _ref3.children,
      dataTest = _ref3.dataTest;
  return /*#__PURE__*/React.createElement(StyledTableBody, {
    dataTest: dataTest
  }, children);
};

var _default = TableBody;
exports.default = _default;
