// @kiwicom/orbit-components/lib/Accordion/AccordionSection/components/SectionHeader.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Button = _interopRequireDefault(require("../../../Button"));

var _defaultTheme = _interopRequireDefault(require("../../../defaultTheme"));

var _transition = _interopRequireDefault(require("../../../utils/transition"));

var Wrapper = _styledComponents.default.div.withConfig({
  displayName: "SectionHeader__Wrapper",
  componentId: "sc-1p5j5w5-0"
})(["display:flex;padding:", ";background-color:", ";align-items:center;min-height:", ";transition:", ";"], function (_ref) {
  var theme = _ref.theme,
      noPadding = _ref.noPadding;
  return !noPadding && theme.orbit.spaceLarge;
}, function (_ref2) {
  var theme = _ref2.theme;
  return theme.orbit.paletteWhite;
}, function (_ref3) {
  var expanded = _ref3.expanded;
  return expanded ? "19px" : "44px";
}, (0, _transition.default)(["min-height"], "fast", "ease-in-out"));

var HeaderContent = _styledComponents.default.div.withConfig({
  displayName: "SectionHeader__HeaderContent",
  componentId: "sc-1p5j5w5-1"
})(["display:flex;flex-grow:1;align-items:center;"]);

var HeaderActions = _styledComponents.default.div.withConfig({
  displayName: "SectionHeader__HeaderActions",
  componentId: "sc-1p5j5w5-2"
})(["display:flex;margin-left:", ";"], function (_ref4) {
  var theme = _ref4.theme;
  return theme.orbit.spaceLarge;
}); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198


Wrapper.defaultProps = {
  theme: _defaultTheme.default
}; // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198

HeaderActions.defaultProps = {
  theme: _defaultTheme.default
};

var AccordionSectionHeader = function AccordionSectionHeader(_ref5) {
  var children = _ref5.children,
      actions = _ref5.actions,
      expanded = _ref5.expanded,
      onExpand = _ref5.onExpand,
      expandable = _ref5.expandable,
      dataTest = _ref5.dataTest;
  return /*#__PURE__*/React.createElement(Wrapper, {
    expanded: expanded,
    "data-test": dataTest && "".concat(dataTest, "Header")
  }, /*#__PURE__*/React.createElement(HeaderContent, null, children), !expanded && /*#__PURE__*/React.createElement(HeaderActions, null, expandable && (actions || /*#__PURE__*/React.createElement(_Button.default, {
    onClick: onExpand,
    type: "secondary"
  }, "Open"))));
};

var _default = AccordionSectionHeader;
exports.default = _default;
