// @kiwicom/orbit-components/lib/deprecated/TripSector/TripDate/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _Calendar = _interopRequireDefault(require("../../../icons/Calendar"));

var _defaultTheme = _interopRequireDefault(require("../../../defaultTheme"));

var _Text = _interopRequireWildcard(require("../../../Text"));

var _Clock = _interopRequireDefault(require("../../../icons/Clock"));

var _Stack = _interopRequireDefault(require("../../../Stack"));

var _Badge = _interopRequireDefault(require("../../../Badge"));

var StyledTripDate = _styledComponents.default.div.withConfig({
  displayName: "TripDate__StyledTripDate",
  componentId: "h285yc-0"
})(["display:flex;align-items:center;margin-bottom:", ";padding-bottom:", ";background:", ";z-index:1;", "{font-weight:", ";height:", ";}"], function (_ref) {
  var theme = _ref.theme;
  return theme.orbit.spaceXSmall;
}, function (_ref2) {
  var theme = _ref2.theme;
  return theme.orbit.spaceXXSmall;
}, function (_ref3) {
  var theme = _ref3.theme;
  return theme.orbit.paletteWhite;
}, _Text.StyledText, function (_ref4) {
  var theme = _ref4.theme;
  return theme.orbit.fontWeightMedium;
}, function (_ref5) {
  var theme = _ref5.theme;
  return theme.orbit.heightIconSmall;
}); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198


StyledTripDate.defaultProps = {
  theme: _defaultTheme.default
};

var TripDate = function TripDate(_ref6) {
  var children = _ref6.children,
      dataTest = _ref6.dataTest,
      duration = _ref6.duration;
  return /*#__PURE__*/React.createElement(StyledTripDate, {
    "data-test": dataTest
  }, /*#__PURE__*/React.createElement(_Stack.default, {
    direction: "row",
    spacing: "XSmall",
    shrink: true
  }, /*#__PURE__*/React.createElement(_Calendar.default, {
    size: "small"
  }), /*#__PURE__*/React.createElement(_Text.default, null, children)), duration && /*#__PURE__*/React.createElement(_Badge.default, {
    icon: /*#__PURE__*/React.createElement(_Clock.default, null)
  }, duration));
};

var _default = TripDate;
exports.default = _default;
