// @kiwicom/orbit-components/lib/Timeline/TimelineStep/components/TimelineStepMobile.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _StyledRelative = _interopRequireDefault(require("../primitives/StyledRelative"));

var _defaultTheme = _interopRequireDefault(require("../../../defaultTheme"));

var _CustomBadge = _interopRequireDefault(require("./CustomBadge"));

var _Badge = _interopRequireDefault(require("../../../Badge"));

var _Text = _interopRequireDefault(require("../../../Text"));

var _Stack = _interopRequireDefault(require("../../../Stack"));

var _StyledIconWrapper = _interopRequireDefault(require("../primitives/StyledIconWrapper"));

var _StyledProgressLine = _interopRequireDefault(require("../primitives/StyledProgressLine"));

var _StyledText = _interopRequireDefault(require("../primitives/StyledText"));

var StyledIndent = _styledComponents.default.div.withConfig({
  displayName: "TimelineStepMobile__StyledIndent",
  componentId: "sc-12kfx34-0"
})(["padding-left:", ";"], function (_ref) {
  var theme = _ref.theme;
  return theme.orbit.spaceXSmall;
}); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198


StyledIndent.defaultProps = {
  theme: _defaultTheme.default
};

var TimelineStepMobile = function TimelineStepMobile(_ref2) {
  var type = _ref2.type,
      subLabel = _ref2.subLabel,
      label = _ref2.label,
      nextType = _ref2.nextType,
      typeIcon = _ref2.typeIcon,
      children = _ref2.children,
      last = _ref2.last;
  return /*#__PURE__*/React.createElement(_StyledRelative.default, null, /*#__PURE__*/React.createElement(_Stack.default, {
    flex: true,
    spaceAfter: "large",
    align: "stretch",
    desktop: {
      align: "start"
    }
  }, /*#__PURE__*/React.createElement(_StyledIconWrapper.default, {
    mobile: true
  }, typeIcon), !last && /*#__PURE__*/React.createElement(_StyledProgressLine.default, {
    status: nextType,
    "data-test": "progressLine"
  }), /*#__PURE__*/React.createElement(_Stack.default, {
    flex: true,
    shrink: true,
    direction: "column",
    spacing: "XSmall"
  }, /*#__PURE__*/React.createElement(_Stack.default, {
    flex: true,
    spacing: "XSmall",
    align: "center"
  }, !type ? /*#__PURE__*/React.createElement(_CustomBadge.default, null, label) : /*#__PURE__*/React.createElement(_Badge.default, {
    type: type
  }, label), subLabel && /*#__PURE__*/React.createElement(_StyledText.default, null, /*#__PURE__*/React.createElement(_Text.default, {
    size: "small"
  }, subLabel))), /*#__PURE__*/React.createElement(StyledIndent, null, /*#__PURE__*/React.createElement(_Text.default, {
    type: type ? "primary" : "secondary"
  }, children)))));
};

var _default = TimelineStepMobile;
exports.default = _default;
