// @kiwicom/orbit-components/es/deprecated/DestinationCard/index.js
import _classCallCheck from "@babel/runtime/helpers/esm/classCallCheck";
import _createClass from "@babel/runtime/helpers/esm/createClass";
import _assertThisInitialized from "@babel/runtime/helpers/esm/assertThisInitialized";
import _inherits from "@babel/runtime/helpers/esm/inherits";
import _possibleConstructorReturn from "@babel/runtime/helpers/esm/possibleConstructorReturn";
import _getPrototypeOf from "@babel/runtime/helpers/esm/getPrototypeOf";
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = _getPrototypeOf(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = _getPrototypeOf(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return _possibleConstructorReturn(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

import * as React from "react";
import styled, { css } from "styled-components";
import Heading from "../../Heading";
import Stack from "../../Stack";
import ArrowUpIcon from "../../icons/ArrowUp";
import defaultTheme from "../../defaultTheme";
import FlightDirectIcon from "../../icons/FlightDirect";
import { BASE_URL, SMALLEST_HEIGHT } from "./consts";
import LazyImage from "../../LazyImage";
import Text from "../../Text";
import randomID from "../../utils/randomID";
import handleKeyDown from "../../utils/handleKeyDown";
var overlayCss = css(["position:absolute;top:0;left:0;z-index:1;width:100%;height:100%;border-radius:", ";transition:opacity ", " ease-in-out;"], function (_ref) {
  var theme = _ref.theme;
  return theme.orbit.borderRadiusNormal;
}, function (_ref2) {
  var theme = _ref2.theme;
  return theme.orbit.durationNormal;
});
var StyledOverlay = styled.div.withConfig({
  displayName: "DestinationCard__StyledOverlay",
  componentId: "mo8kks-0"
})(["", ";opacity:1;background:linear-gradient( to bottom,rgba(0,0,0,0) 0%,rgba(0,0,0,0) 50%,rgba(0,0,0,0.13) 65%,rgba(0,0,0,0.3) 75%,rgba(0,0,0,0.8) 100% );"], overlayCss); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198

StyledOverlay.defaultProps = {
  theme: defaultTheme
};
var StyledOverlayHover = styled.div.withConfig({
  displayName: "DestinationCard__StyledOverlayHover",
  componentId: "mo8kks-1"
})(["", ";opacity:0;background:rgba(0,0,0,0.8);"], overlayCss); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198

StyledOverlayHover.defaultProps = {
  theme: defaultTheme
};
var StyledDestinationCardContent = styled.div.withConfig({
  displayName: "DestinationCard__StyledDestinationCardContent",
  componentId: "mo8kks-2"
})(["position:relative;z-index:2;display:flex;flex-direction:column;padding:", ";width:100%;transition:bottom ", " ease-in-out;bottom:", ";"], function (_ref3) {
  var theme = _ref3.theme;
  return theme.orbit.spaceSmall;
}, function (_ref4) {
  var theme = _ref4.theme;
  return theme.orbit.durationFast;
}, function (_ref5) {
  var hiddenContentHeight = _ref5.hiddenContentHeight,
      theme = _ref5.theme;
  return "-".concat(hiddenContentHeight + parseInt(theme.orbit.spaceSmall, 10), "px");
}); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198

StyledDestinationCardContent.defaultProps = {
  theme: defaultTheme
};
var StyledDestinationCardHeader = styled.div.withConfig({
  displayName: "DestinationCard__StyledDestinationCardHeader",
  componentId: "mo8kks-3"
})(["margin-bottom:", ";"], function (_ref6) {
  var theme = _ref6.theme;
  return theme.orbit.spaceXSmall;
}); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198

StyledDestinationCardHeader.defaultProps = {
  theme: defaultTheme
};
var Shown = styled.div.withConfig({
  displayName: "DestinationCard__Shown",
  componentId: "mo8kks-4"
})(["transition:all ", " ease-in-out;opacity:0;position:relative;top:\"-50%\";margin-bottom:", ";"], function (_ref7) {
  var theme = _ref7.theme;
  return theme.orbit.durationFast;
}, function (_ref8) {
  var theme = _ref8.theme;
  return theme.orbit.spaceXXSmall;
}); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198

Shown.defaultProps = {
  theme: defaultTheme
};
var ArrowUp = styled(ArrowUpIcon).withConfig({
  displayName: "DestinationCard__ArrowUp",
  componentId: "mo8kks-5"
})(["transform:rotate(90deg);margin-left:", ";"], function (_ref9) {
  var theme = _ref9.theme;
  return theme.orbit.spaceXXSmall;
}); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198

ArrowUp.defaultProps = {
  theme: defaultTheme
};
var StyledDestinationCardHiddenContent = styled.div.withConfig({
  displayName: "DestinationCard__StyledDestinationCardHiddenContent",
  componentId: "mo8kks-6"
})(["padding:", " 0 0;border-top:1px solid rgba(255,255,255,0.3);"], function (_ref10) {
  var theme = _ref10.theme;
  return theme.orbit.spaceSmall;
}); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198

StyledDestinationCardHiddenContent.defaultProps = {
  theme: defaultTheme
};
var FlightDirect = styled(FlightDirectIcon).withConfig({
  displayName: "DestinationCard__FlightDirect",
  componentId: "mo8kks-7"
})(["transform:", ";"], function (_ref11) {
  var isReturn = _ref11.isReturn;
  return isReturn && "rotate(180deg)";
});
var StyledDestination = styled.div.withConfig({
  displayName: "DestinationCard__StyledDestination",
  componentId: "mo8kks-8"
})(["position:relative;transition:top ", " ease-in-out;top:16px;"], function (_ref12) {
  var theme = _ref12.theme;
  return theme.orbit.durationFast;
}); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198

StyledDestination.defaultProps = {
  theme: defaultTheme
};
var StyledDestinationCard = styled(function (_ref13) {
  var height = _ref13.height,
      imageURL = _ref13.imageURL,
      theme = _ref13.theme,
      props = _objectWithoutProperties(_ref13, ["height", "imageURL", "theme"]);

  return /*#__PURE__*/React.createElement("div", props);
}).withConfig({
  displayName: "DestinationCard__StyledDestinationCard",
  componentId: "mo8kks-9"
})(["width:100%;height:", ";display:flex;align-items:flex-end;position:relative;box-sizing:border-box;border-radius:", ";overflow:hidden;cursor:pointer;&:hover{", "{opacity:0;}", "{top:0;}", "{opacity:1;}", "{bottom:0;}", "{opacity:1;top:0;}}&:focus{outline:none;box-shadow:0 0 1px 1px ", ",0 0 1px 3px rgba(1,118,210,0.6);}"], function (_ref14) {
  var height = _ref14.height;
  return height ? "".concat(height, "px") : "100%";
}, function (_ref15) {
  var theme = _ref15.theme;
  return theme.orbit.borderRadiusNormal;
}, StyledOverlay, StyledDestination, StyledOverlayHover, StyledDestinationCardContent, Shown, function (_ref16) {
  var theme = _ref16.theme;
  return theme.orbit.colorTextButtonWhiteBordered;
}); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198

StyledDestinationCard.defaultProps = {
  theme: defaultTheme
};

var SmallHeading = function SmallHeading(_ref17) {
  var children = _ref17.children;
  return /*#__PURE__*/React.createElement(Text, {
    type: "white",
    as: "div",
    size: "small",
    weight: "bold"
  }, children);
};

var DestinationCard = /*#__PURE__*/function (_React$PureComponent) {
  _inherits(DestinationCard, _React$PureComponent);

  var _super = _createSuper(DestinationCard);

  function DestinationCard() {
    var _this;

    _classCallCheck(this, DestinationCard);

    for (var _len = arguments.length, args = new Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    _this = _super.call.apply(_super, [this].concat(args));

    _defineProperty(_assertThisInitialized(_this), "state", {
      hiddenContentHeight: 0
    });

    _defineProperty(_assertThisInitialized(_this), "hiddenContent", /*#__PURE__*/React.createRef());

    _defineProperty(_assertThisInitialized(_this), "cardID", randomID("destinationCardID"));

    _defineProperty(_assertThisInitialized(_this), "setHeight", function () {
      var _assertThisInitialize = _assertThisInitialized(_this),
          hiddenContent = _assertThisInitialize.hiddenContent;

      if (hiddenContent.current) {
        _this.setState({
          hiddenContentHeight: hiddenContent.current.clientHeight
        });
      }
    });

    return _this;
  }

  _createClass(DestinationCard, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      this.setHeight();
    }
  }, {
    key: "componentDidUpdate",
    value: function componentDidUpdate(prevProps) {
      if (prevProps !== this.props) {
        this.setHeight();
      }
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props = this.props,
          dataTest = _this$props.dataTest,
          departureCity = _this$props.departureCity,
          destinationCity = _this$props.destinationCity,
          destinationCountry = _this$props.destinationCountry,
          price = _this$props.price,
          timeOfStay = _this$props.timeOfStay,
          outbound = _this$props.outbound,
          inbound = _this$props.inbound,
          _this$props$height = _this$props.height,
          height = _this$props$height === void 0 ? 300 : _this$props$height,
          onClick = _this$props.onClick,
          _this$props$tabIndex = _this$props.tabIndex,
          tabIndex = _this$props$tabIndex === void 0 ? "0" : _this$props$tabIndex;
      var hiddenContentHeight = this.state.hiddenContentHeight;
      var image = this.props.image.toLowerCase();
      return /*#__PURE__*/React.createElement(StyledDestinationCard, {
        "data-test": dataTest,
        onClick: onClick,
        onKeyDown: handleKeyDown(onClick),
        height: height >= SMALLEST_HEIGHT ? height : SMALLEST_HEIGHT,
        tabIndex: tabIndex,
        role: "link",
        "aria-labelledby": this.cardID
      }, /*#__PURE__*/React.createElement(LazyImage, {
        original: {
          webp: " ".concat(BASE_URL, "/photos/385x320/").concat(image, ".webp"),
          jpg: "".concat(BASE_URL, "/photos/385x320/").concat(image, ".jpg")
        },
        placeholder: {
          webp: " ".concat(BASE_URL, "/photos/30x30/").concat(image, ".webp"),
          jpg: "".concat(BASE_URL, "/photos/30x30/").concat(image, ".jpg")
        } // Add that to images kiwi
        ,
        name: destinationCity
      }), /*#__PURE__*/React.createElement(StyledOverlay, null), /*#__PURE__*/React.createElement(StyledOverlayHover, null), /*#__PURE__*/React.createElement(StyledDestinationCardContent, {
        hiddenContentHeight: hiddenContentHeight
      }, /*#__PURE__*/React.createElement(StyledDestinationCardHeader, {
        id: this.cardID
      }, /*#__PURE__*/React.createElement(Shown, null, /*#__PURE__*/React.createElement(Stack, {
        flex: true,
        align: "center",
        justify: "start",
        spacing: "XXXSmall"
      }, /*#__PURE__*/React.createElement(Heading, {
        type: "title3",
        inverted: true
      }, departureCity), /*#__PURE__*/React.createElement(ArrowUp, {
        customColor: "#fff",
        size: "small"
      }))), /*#__PURE__*/React.createElement(StyledDestination, null, /*#__PURE__*/React.createElement(Heading, {
        type: "title1",
        inverted: true
      }, destinationCity)), /*#__PURE__*/React.createElement(Shown, null, /*#__PURE__*/React.createElement(Heading, {
        type: "title4",
        inverted: true
      }, destinationCountry))), /*#__PURE__*/React.createElement(Stack, {
        flex: true,
        justify: "between",
        spaceAfter: "small",
        align: "end"
      }, /*#__PURE__*/React.createElement(Heading, {
        type: "title3",
        inverted: true
      }, price), timeOfStay && /*#__PURE__*/React.createElement(Heading, {
        type: "title4",
        inverted: true
      }, timeOfStay)), /*#__PURE__*/React.createElement(StyledDestinationCardHiddenContent, {
        ref: this.hiddenContent
      }, /*#__PURE__*/React.createElement(Stack, {
        direction: "row",
        justify: "between"
      }, /*#__PURE__*/React.createElement(SmallHeading, null, outbound.text ? outbound.text : /*#__PURE__*/React.createElement(React.Fragment, null, /*#__PURE__*/React.createElement(FlightDirect, {
        size: "small"
      }), outbound.date)), /*#__PURE__*/React.createElement(SmallHeading, null, outbound.type, "\xA0\u2022\xA0", outbound.duration)), inbound && /*#__PURE__*/React.createElement(Stack, {
        direction: "row",
        justify: "between"
      }, /*#__PURE__*/React.createElement(SmallHeading, null, /*#__PURE__*/React.createElement(FlightDirect, {
        size: "small",
        isReturn: true
      }), inbound.date), /*#__PURE__*/React.createElement(SmallHeading, null, inbound.type, "\xA0\u2022\xA0", inbound.duration)))));
    }
  }]);

  return DestinationCard;
}(React.PureComponent);

export default DestinationCard;
