// @kiwicom/orbit-components/lib/Table/TableFooter/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.StyledTableFooter = void 0;

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _defaultTheme = _interopRequireDefault(require("../../defaultTheme"));

var StyledTableFooter = (0, _styledComponents.default)(function (_ref) {
  var children = _ref.children,
      className = _ref.className,
      dataTest = _ref.dataTest;
  return /*#__PURE__*/React.createElement("tfoot", {
    className: className,
    "data-test": dataTest
  }, children);
}).withConfig({
  displayName: "TableFooter__StyledTableFooter",
  componentId: "uyl8kl-0"
})(["width:100%;white-space:nowrap;background:", ";font-weight:bold;"], function (_ref2) {
  var theme = _ref2.theme;
  return theme.orbit.paletteCloudDark;
}); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198

exports.StyledTableFooter = StyledTableFooter;
StyledTableFooter.defaultProps = {
  theme: _defaultTheme.default
};

var TableFooter = function TableFooter(_ref3) {
  var children = _ref3.children,
      dataTest = _ref3.dataTest;
  return /*#__PURE__*/React.createElement(StyledTableFooter, {
    dataTest: dataTest
  }, children);
};

var _default = TableFooter;
exports.default = _default;
