// @kiwicom/orbit-components/lib/Table/TableHead/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _defaultTheme = _interopRequireDefault(require("../../defaultTheme"));

var _index = require("../TableCell/index");

var StyledTableHead = (0, _styledComponents.default)(function (_ref) {
  var children = _ref.children,
      className = _ref.className,
      dataTest = _ref.dataTest;
  return /*#__PURE__*/React.createElement("thead", {
    className: className,
    "data-test": dataTest
  }, children);
}).withConfig({
  displayName: "TableHead__StyledTableHead",
  componentId: "sc-17bka47-0"
})(["border-bottom:1px solid ", ";width:100%;white-space:nowrap;", "{font-weight:", ";}"], function (_ref2) {
  var theme = _ref2.theme;
  return theme.orbit.borderColorTableHead;
}, _index.StyledTableCell, function (_ref3) {
  var theme = _ref3.theme;
  return theme.orbit.fontWeightTableHead;
}); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198

StyledTableHead.defaultProps = {
  theme: _defaultTheme.default
};

var TableHead = function TableHead(_ref4) {
  var children = _ref4.children,
      dataTest = _ref4.dataTest;
  return /*#__PURE__*/React.createElement(StyledTableHead, {
    dataTest: dataTest
  }, children);
};

var _default = TableHead;
exports.default = _default;
