// @kiwicom/orbit-components/es/Table/TableHead/index.js
import * as React from "react";
import styled from "styled-components";
import defaultTheme from "../../defaultTheme";
import { StyledTableCell } from "../TableCell/index";
var StyledTableHead = styled(function (_ref) {
  var children = _ref.children,
      className = _ref.className,
      dataTest = _ref.dataTest;
  return /*#__PURE__*/React.createElement("thead", {
    className: className,
    "data-test": dataTest
  }, children);
}).withConfig({
  displayName: "TableHead__StyledTableHead",
  componentId: "sc-17bka47-0"
})(["border-bottom:1px solid ", ";width:100%;white-space:nowrap;", "{font-weight:", ";}"], function (_ref2) {
  var theme = _ref2.theme;
  return theme.orbit.borderColorTableHead;
}, StyledTableCell, function (_ref3) {
  var theme = _ref3.theme;
  return theme.orbit.fontWeightTableHead;
}); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198

StyledTableHead.defaultProps = {
  theme: defaultTheme
};

var TableHead = function TableHead(_ref4) {
  var children = _ref4.children,
      dataTest = _ref4.dataTest;
  return /*#__PURE__*/React.createElement(StyledTableHead, {
    dataTest: dataTest
  }, children);
};

export default TableHead;
