// @kiwicom/orbit-components/es/Table/TableFooter/index.js
import * as React from "react";
import styled from "styled-components";
import defaultTheme from "../../defaultTheme";
export var StyledTableFooter = styled(function (_ref) {
  var children = _ref.children,
      className = _ref.className,
      dataTest = _ref.dataTest;
  return /*#__PURE__*/React.createElement("tfoot", {
    className: className,
    "data-test": dataTest
  }, children);
}).withConfig({
  displayName: "TableFooter__StyledTableFooter",
  componentId: "uyl8kl-0"
})(["width:100%;white-space:nowrap;background:", ";font-weight:bold;"], function (_ref2) {
  var theme = _ref2.theme;
  return theme.orbit.paletteCloudDark;
}); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198

StyledTableFooter.defaultProps = {
  theme: defaultTheme
};

var TableFooter = function TableFooter(_ref3) {
  var children = _ref3.children,
      dataTest = _ref3.dataTest;
  return /*#__PURE__*/React.createElement(StyledTableFooter, {
    dataTest: dataTest
  }, children);
};

export default TableFooter;
