// @kiwicom/orbit-components/es/Modal/ModalContext.js
import _extends from "@babel/runtime/helpers/esm/extends";
import * as React from "react";
export var ModalContext = /*#__PURE__*/React.createContext({
  setHasModalTitle: function setHasModalTitle() {},
  setHasModalSection: function setHasModalSection() {},
  removeHasModalSection: function removeHasModalSection() {},
  callContextFunctions: function callContextFunctions() {},
  hasModalSection: false,
  isMobileFullPage: false,
  isInsideModal: false,
  closable: false
});
ModalContext.displayName = "ModalOrbitContext"; // $FlowFixMe

export var withModalContext = function withModalContext(Component) {
  return function (props) {
    return /*#__PURE__*/React.createElement(ModalContext.Consumer, null,
    /* $FlowFixMe(>=0.115.0) This comment suppresses an error found when upgrading Flow
     * to v0.115.0. To view the error, delete this comment and run Flow. */
    function (contextProps) {
      return /*#__PURE__*/React.createElement(Component, _extends({}, props, contextProps));
    });
  };
};
export var useModalContext = function useModalContext() {
  return React.useContext(ModalContext);
};
