// @kiwicom/orbit-components/es/Hide/index.js
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";
import * as React from "react";
import styled from "styled-components";
import defaultTheme from "../defaultTheme";
import getViewportHideStyles from "./helpers/getViewportHideStyles";
import getDisplay from "./helpers/getDisplay";
var StyledHide = styled(function (_ref) {
  var on = _ref.on,
      block = _ref.block,
      theme = _ref.theme,
      props = _objectWithoutProperties(_ref, ["on", "block", "theme"]);

  return /*#__PURE__*/React.createElement("div", props);
}).withConfig({
  displayName: "Hide__StyledHide",
  componentId: "ji2pgq-0"
})(["", ";"], function (_ref2) {
  var on = _ref2.on;
  return getViewportHideStyles(on, getDisplay);
}); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198

StyledHide.defaultProps = {
  theme: defaultTheme
};

var Hide = function Hide(_ref3) {
  var _ref3$on = _ref3.on,
      on = _ref3$on === void 0 ? [] : _ref3$on,
      block = _ref3.block,
      children = _ref3.children;
  return /*#__PURE__*/React.createElement(StyledHide, {
    on: on,
    block: block
  }, children);
};

export default Hide;
