// eslint-plugin-orbit-components/dist/rules/noCustomColors.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;
var _default = {
  meta: {
    type: "suggestion",
    docs: {
      description: "Prevents inconsistencies between Orbit and custom written colors",
      category: "Possible Errors",
      recommended: true
    }
  },
  create: context => {
    const allowed = ["background", "background-color", "color"];
    return {
      TemplateElement(node) {
        if (node.value.cooked) {
          var _node$value$cooked;

          const properties = (_node$value$cooked = node.value.cooked) === null || _node$value$cooked === void 0 ? void 0 : _node$value$cooked.replace(/\n/gm, "").trim().split(";").map(v => v.trim()).filter(Boolean).map(p => p.split(":")).filter(([n]) => allowed.includes(n));
          properties.forEach(p => {
            if (p[1]) {
              context.report({
                // @ts-expect-error TODO
                node,
                message: "".concat(p[0], " should be replaced with Orbit design token")
              });
            }
          });
        }
      }

    };
  }
};
exports.default = _default;
