// @kiwicom/orbit-components/lib/deprecated/TripSector/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "TripDate", {
  enumerable: true,
  get: function get() {
    return _TripDate.default;
  }
});
Object.defineProperty(exports, "TripLayover", {
  enumerable: true,
  get: function get() {
    return _TripLayover.default;
  }
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _TripSegment = require("../TripSegment");

var _defaultTheme = _interopRequireDefault(require("../../defaultTheme"));

var _rtl = require("../../utils/rtl");

var _TripDate = _interopRequireDefault(require("./TripDate"));

var _TripLayover = _interopRequireDefault(require("./TripLayover"));

var StyledTripSector = _styledComponents.default.div.withConfig({
  displayName: "TripSector__StyledTripSector",
  componentId: "wez4be-0"
})(["display:flex;position:relative;width:100%;padding-", ":2px;box-sizing:border-box;flex-direction:column;font-family:", ";&:before{content:\" \";display:block;position:absolute;background-color:", ";width:1px;height:100%;", ":7px;top:0;}", ":last-child{position:relative;", ":after{content:\" \";position:absolute;top:13px;right:0;bottom:0;left:0;width:16px;background:", ";}}"], _rtl.right, function (_ref) {
  var theme = _ref.theme;
  return theme.orbit.fontFamily;
}, function (_ref2) {
  var theme = _ref2.theme;
  return theme.orbit.paletteCloudNormal;
}, _rtl.left, _TripSegment.StyledTripSegment, _TripSegment.StyledTripSegmentMilestone, function (_ref3) {
  var theme = _ref3.theme;
  return theme.orbit.paletteWhite;
}); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198


StyledTripSector.defaultProps = {
  theme: _defaultTheme.default
};

var TripSector = function TripSector(_ref4) {
  var children = _ref4.children,
      dataTest = _ref4.dataTest;
  return /*#__PURE__*/React.createElement(StyledTripSector, {
    "data-test": dataTest
  }, children);
};

var _default = TripSector;
exports.default = _default;
