// @kiwicom/orbit-components/lib/SmartPassIllustrations/illustrations/IllustrationWrapper.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var React = _interopRequireWildcard(require("react"));

var _defaultTheme = _interopRequireDefault(require("../../defaultTheme"));

var _consts = require("../../primitives/IllustrationPrimitive/consts");

var _tokens;

var tokens = (_tokens = {}, (0, _defineProperty2.default)(_tokens, _consts.SIZE_OPTIONS.EXTRASMALL, _defaultTheme.default.orbit.heightIllustrationSmall), (0, _defineProperty2.default)(_tokens, _consts.SIZE_OPTIONS.SMALL, "120px"), (0, _defineProperty2.default)(_tokens, _consts.SIZE_OPTIONS.MEDIUM, _defaultTheme.default.orbit.heightIllustrationMedium), (0, _defineProperty2.default)(_tokens, _consts.SIZE_OPTIONS.LARGE, "280px"), (0, _defineProperty2.default)(_tokens, _consts.SIZE_OPTIONS.DISPLAY, "460px"), _tokens);

var IllustrationWrapper = function IllustrationWrapper(_ref) {
  var _ref$size = _ref.size,
      size = _ref$size === void 0 ? "medium" : _ref$size,
      viewBox = _ref.viewBox,
      title = _ref.title,
      description = _ref.description,
      ariaLabelledby = _ref.ariaLabelledby,
      dataTest = _ref.dataTest,
      children = _ref.children;
  var height = tokens[size];
  return /*#__PURE__*/React.createElement("svg", {
    height: height,
    "data-test": dataTest,
    viewBox: viewBox,
    fill: "none",
    role: "img",
    "aria-labelledby": ariaLabelledby,
    xmlns: "http://www.w3.org/2000/svg"
  }, /*#__PURE__*/React.createElement("title", {
    id: ariaLabelledby
  }, title), /*#__PURE__*/React.createElement("desc", null, description), children);
};

var _default = IllustrationWrapper;
exports.default = _default;
