// @kiwicom/orbit-components/es/primitives/ButtonPrimitive/components/ButtonPrimitiveContentChildren.js
import * as React from "react";
import styled from "styled-components";
import { left } from "../../../utils/rtl";
import defaultTheme from "../../../defaultTheme";
var StyledButtonPrimitiveContentChildren = styled.div.withConfig({
  displayName: "ButtonPrimitiveContentChildren__StyledButtonPrimitiveContentChildren",
  componentId: "xn0ytv-0"
})(["display:inline-block;width:", ";text-align:", ";"], function (_ref) {
  var contentWidth = _ref.contentWidth;
  return contentWidth;
}, function (_ref2) {
  var hasIcon = _ref2.hasIcon;
  return hasIcon && left;
}); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198

StyledButtonPrimitiveContentChildren.defaultProps = {
  theme: defaultTheme
};

var ButtonPrimitiveContentChildren = function ButtonPrimitiveContentChildren(_ref3) {
  var children = _ref3.children,
      hasIcon = _ref3.hasIcon,
      contentWidth = _ref3.contentWidth;
  return /*#__PURE__*/React.createElement(StyledButtonPrimitiveContentChildren, {
    hasIcon: hasIcon,
    contentWidth: contentWidth
  }, children);
};

export default ButtonPrimitiveContentChildren;
