// @kiwicom/orbit-components/lib/Layout/consts.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.LAYOUT_SETTINGS = exports.LAYOUT_OPTIONS = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _LAYOUT_SETTINGS;

var LAYOUT_OPTIONS = {
  SEARCH: "Search",
  BOOKING: "Booking",
  MMB: "MMB"
};
exports.LAYOUT_OPTIONS = LAYOUT_OPTIONS;
var LAYOUT_SETTINGS = (_LAYOUT_SETTINGS = {}, (0, _defineProperty2.default)(_LAYOUT_SETTINGS, LAYOUT_OPTIONS.SEARCH, {
  columns: "1fr",
  columnGap: "16px",
  maxWidth: "1440px",
  tablet: {
    columns: "minmax(256px, 3fr) 7fr"
  },
  desktop: {
    columnGap: "24px"
  },
  largeDesktop: {
    columns: "256px 1fr 288px"
  },
  layoutColumns: {
    // $FlowIssue
    0: {
      hideOn: ["smallMobile", "mediumMobile", "largeMobile"],
      as: "aside"
    },
    // $FlowIssue
    1: {
      as: "main"
    },
    // $FlowIssue
    2: {
      hideOn: ["smallMobile", "mediumMobile", "largeMobile", "tablet", "desktop"]
    }
  }
}), (0, _defineProperty2.default)(_LAYOUT_SETTINGS, LAYOUT_OPTIONS.BOOKING, {
  columns: "1fr",
  columnGap: "16px",
  maxWidth: "1200px",
  tablet: {
    columns: "7fr minmax(272px, 3fr)"
  },
  desktop: {
    columnGap: "24px"
  },
  layoutColumns: {
    // $FlowIssue
    0: {
      as: "main"
    },
    // $FlowIssue
    1: {
      as: "aside"
    }
  }
}), (0, _defineProperty2.default)(_LAYOUT_SETTINGS, LAYOUT_OPTIONS.MMB, {
  columns: "1fr",
  columnGap: "16px",
  maxWidth: "960px",
  desktop: {
    columnGap: "24px"
  },
  layoutColumns: {
    // $FlowIssue
    0: {
      as: "main"
    }
  }
}), _LAYOUT_SETTINGS);
exports.LAYOUT_SETTINGS = LAYOUT_SETTINGS;
var _default = LAYOUT_SETTINGS;
exports.default = _default;
