// @kiwicom/orbit-components/lib/Layout/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "LayoutColumn", {
  enumerable: true,
  get: function get() {
    return _LayoutColumn.default;
  }
});
exports.default = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _defaultTheme = _interopRequireDefault(require("../defaultTheme"));

var _Grid = _interopRequireDefault(require("../utils/Grid"));

var _consts = require("./consts");

var _mediaQuery = _interopRequireDefault(require("../utils/mediaQuery"));

var _LayoutColumn = _interopRequireDefault(require("./LayoutColumn"));

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2.default)(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var getChildrenProps = function getChildrenProps(type, key) {
  if (_consts.LAYOUT_SETTINGS[type].layoutColumns && _consts.LAYOUT_SETTINGS[type].layoutColumns[key]) {
    return _consts.LAYOUT_SETTINGS[type].layoutColumns[key];
  }

  return null;
};

var StyledLayout = (0, _styledComponents.default)(_Grid.default).withConfig({
  displayName: "Layout__StyledLayout",
  componentId: "vx9wse-0"
})(["box-sizing:border-box;margin:0 auto;width:100%;padding:0 ", ";", ";"], function (_ref) {
  var theme = _ref.theme;
  return theme.orbit.spaceMedium;
}, _mediaQuery.default.desktop((0, _styledComponents.css)(["padding:", ";"], function (_ref2) {
  var theme = _ref2.theme;
  return theme.orbit.spaceLarge;
}))); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198

StyledLayout.defaultProps = {
  theme: _defaultTheme.default
};

var Layout = function Layout(_ref3) {
  var children = _ref3.children,
      type = _ref3.type,
      dataTest = _ref3.dataTest;
  return /*#__PURE__*/React.createElement(StyledLayout, (0, _extends2.default)({}, _consts.LAYOUT_SETTINGS[type], {
    dataTest: dataTest
  }), React.Children.map(children, function (item, key) {
    return /*#__PURE__*/React.cloneElement(item, _objectSpread(_objectSpread({}, getChildrenProps(type, key)), item.props));
  }));
};

var _default = Layout;
exports.default = _default;
