// @kiwicom/orbit-components/es/ButtonLink/helpers/getButtonLinkStyles.js
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

import { TOKENS, TYPES } from "../consts";
import getButtonLinkTypeToken from "./getButtonLinkTypeToken";

var getButtonLinkStyles = function getButtonLinkStyles(_ref) {
  var type = _ref.type,
      theme = _ref.theme,
      compact = _ref.compact;

  var wrappedTypeToken = function wrappedTypeToken(name) {
    return getButtonLinkTypeToken(name, type, theme);
  };

  var commonStyles = {
    background: wrappedTypeToken(TOKENS.background),
    foreground: wrappedTypeToken(TOKENS.foreground),
    foregroundHover: wrappedTypeToken(TOKENS.foregroundHover),
    foregroundActive: wrappedTypeToken(TOKENS.foregroundActive),
    foregroundFocus: wrappedTypeToken(TOKENS.foregroundActive)
  };

  if (compact) {
    if (type === TYPES.SECONDARY) {
      var wrappedPrimaryType = function wrappedPrimaryType(name) {
        return getButtonLinkTypeToken(name, TYPES.PRIMARY, theme);
      };

      return _objectSpread(_objectSpread({}, commonStyles), {}, {
        foregroundHover: wrappedPrimaryType(TOKENS.foregroundHover),
        foregroundActive: wrappedPrimaryType(TOKENS.foregroundActive),
        foregroundFocus: wrappedPrimaryType(TOKENS.foregroundActive)
      });
    }

    return commonStyles;
  }

  return _objectSpread({
    backgroundHover: wrappedTypeToken(TOKENS.backgroundHover),
    backgroundActive: wrappedTypeToken(TOKENS.backgroundActive),
    backgroundFocus: wrappedTypeToken(TOKENS.backgroundActive)
  }, commonStyles);
};

export default getButtonLinkStyles;
