// @kiwicom/orbit-components/lib/deprecated/Tile/TileExpandable/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _classCallCheck2 = _interopRequireDefault(require("@babel/runtime/helpers/classCallCheck"));

var _createClass2 = _interopRequireDefault(require("@babel/runtime/helpers/createClass"));

var _assertThisInitialized2 = _interopRequireDefault(require("@babel/runtime/helpers/assertThisInitialized"));

var _inherits2 = _interopRequireDefault(require("@babel/runtime/helpers/inherits"));

var _possibleConstructorReturn2 = _interopRequireDefault(require("@babel/runtime/helpers/possibleConstructorReturn"));

var _getPrototypeOf2 = _interopRequireDefault(require("@babel/runtime/helpers/getPrototypeOf"));

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _Slide = _interopRequireDefault(require("../../../utils/Slide"));

var _defaultTheme = _interopRequireDefault(require("../../../defaultTheme"));

function _createSuper(Derived) { var hasNativeReflectConstruct = _isNativeReflectConstruct(); return function _createSuperInternal() { var Super = (0, _getPrototypeOf2.default)(Derived), result; if (hasNativeReflectConstruct) { var NewTarget = (0, _getPrototypeOf2.default)(this).constructor; result = Reflect.construct(Super, arguments, NewTarget); } else { result = Super.apply(this, arguments); } return (0, _possibleConstructorReturn2.default)(this, result); }; }

function _isNativeReflectConstruct() { if (typeof Reflect === "undefined" || !Reflect.construct) return false; if (Reflect.construct.sham) return false; if (typeof Proxy === "function") return true; try { Boolean.prototype.valueOf.call(Reflect.construct(Boolean, [], function () {})); return true; } catch (e) { return false; } }

var StyledTileExpandable = _styledComponents.default.div.withConfig({
  displayName: "TileExpandable__StyledTileExpandable",
  componentId: "sc-19849bk-0"
})(["border-top:", ";padding:", ";margin:0 ", ";", ";font-size:", ";line-height:", ";color:", ";"], function (_ref) {
  var theme = _ref.theme,
      expanded = _ref.expanded;
  return expanded ? "1px solid ".concat(theme.orbit.paletteCloudNormal) : "0px solid ".concat(theme.orbit.paletteCloudNormal);
}, function (_ref2) {
  var theme = _ref2.theme,
      expanded = _ref2.expanded;
  return expanded && "".concat(theme.orbit.spaceMedium, " 0");
}, function (_ref3) {
  var theme = _ref3.theme;
  return theme.orbit.spaceMedium;
}, function (_ref4) {
  var initialExpanded = _ref4.initialExpanded,
      theme = _ref4.theme;
  return !initialExpanded && (0, _styledComponents.css)(["transition:max-height ", " ease-in-out,padding ", " ease-in-out,border-top ", " ease-in-out;"], theme.orbit.durationFast, theme.orbit.durationFast, theme.orbit.durationFast);
}, function (_ref5) {
  var theme = _ref5.theme;
  return theme.orbit.fontSizeTextNormal;
}, function (_ref6) {
  var theme = _ref6.theme;
  return theme.orbit.lineHeightTextNormal;
}, function (_ref7) {
  var theme = _ref7.theme;
  return theme.orbit.colorTextPrimary;
}); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198


StyledTileExpandable.defaultProps = {
  theme: _defaultTheme.default
};

var TileExpandable = /*#__PURE__*/function (_React$PureComponent) {
  (0, _inherits2.default)(TileExpandable, _React$PureComponent);

  var _super = _createSuper(TileExpandable);

  function TileExpandable(props) {
    var _this;

    (0, _classCallCheck2.default)(this, TileExpandable);
    _this = _super.call(this, props);
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "timeout", void 0);
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "node", void 0);
    (0, _defineProperty2.default)((0, _assertThisInitialized2.default)(_this), "setHeight", function () {
      var _this$node$current;

      var contentHeight = (_this$node$current = _this.node.current) === null || _this$node$current === void 0 ? void 0 : _this$node$current.clientHeight;

      _this.setState({
        contentHeight: contentHeight
      });
    });
    _this.node = /*#__PURE__*/React.createRef();
    _this.state = {
      contentHeight: 0
    };
    return _this;
  }

  (0, _createClass2.default)(TileExpandable, [{
    key: "componentDidMount",
    value: function componentDidMount() {
      this.timeout = setTimeout(this.setHeight, 250); // Prevent showing children on initial render

      window.addEventListener("resize", this.setHeight);
    }
  }, {
    key: "componentDidUpdate",
    value: function componentDidUpdate() {
      // Calculate height on expand - for dynamic components like TripSector
      this.setHeight();
    }
  }, {
    key: "componentWillUnmount",
    value: function componentWillUnmount() {
      if (this.timeout) {
        clearTimeout(this.timeout);
      }

      window.removeEventListener("resize", this.setHeight);
    }
  }, {
    key: "render",
    value: function render() {
      var _this$props = this.props,
          expanded = _this$props.expanded,
          children = _this$props.children,
          initialExpanded = _this$props.initialExpanded;
      return /*#__PURE__*/React.createElement(StyledTileExpandable, {
        expanded: expanded,
        contentHeight: this.state.contentHeight,
        initialExpanded: initialExpanded
      }, /*#__PURE__*/React.createElement(_Slide.default, {
        maxHeight: this.state.contentHeight,
        expanded: expanded
      }, /*#__PURE__*/React.createElement("div", {
        ref: this.node
      }, children)));
    }
  }]);
  return TileExpandable;
}(React.PureComponent);

var _default = TileExpandable;
exports.default = _default;
