// @kiwicom/orbit-components/lib/Hide/index.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _defaultTheme = _interopRequireDefault(require("../defaultTheme"));

var _getViewportHideStyles = _interopRequireDefault(require("./helpers/getViewportHideStyles"));

var _getDisplay = _interopRequireDefault(require("./helpers/getDisplay"));

var StyledHide = (0, _styledComponents.default)(function (_ref) {
  var on = _ref.on,
      block = _ref.block,
      theme = _ref.theme,
      props = (0, _objectWithoutProperties2.default)(_ref, ["on", "block", "theme"]);
  return /*#__PURE__*/React.createElement("div", props);
}).withConfig({
  displayName: "Hide__StyledHide",
  componentId: "ji2pgq-0"
})(["", ";"], function (_ref2) {
  var on = _ref2.on;
  return (0, _getViewportHideStyles.default)(on, _getDisplay.default);
}); // $FlowFixMe: https://github.com/flow-typed/flow-typed/issues/3653#issuecomment-568539198

StyledHide.defaultProps = {
  theme: _defaultTheme.default
};

var Hide = function Hide(_ref3) {
  var _ref3$on = _ref3.on,
      on = _ref3$on === void 0 ? [] : _ref3$on,
      block = _ref3.block,
      children = _ref3.children;
  return /*#__PURE__*/React.createElement(StyledHide, {
    on: on,
    block: block
  }, children);
};

var _default = Hide;
exports.default = _default;
