// @kiwicom/orbit-components/es/Button/consts.js
export var TYPE_OPTIONS = {
  PRIMARY: "primary",
  SECONDARY: "secondary",
  CRITICAL: "critical",
  WHITE: "white",
  PRIMARY_SUBTLE: "primarySubtle",
  CRITICAL_SUBTLE: "criticalSubtle"
};
export var SIZE_OPTIONS = {
  SMALL: "small",
  NORMAL: "normal",
  LARGE: "large"
};
export var TOKENS = {
  // Size tokens
  heightButton: "heightButton",
  loadingWidth: "loadingWidth",
  loadingHeight: "loadingHeight",
  fontSizeButton: "fontSizeButton",
  paddingButton: "paddingButton",
  paddingButtonWithIcons: "paddingButtonWithIcons",
  paddingButtonWithLeftIcon: "paddingButtonWithLeftIcon",
  paddingButtonWithRightIcon: "paddingButtonWithRightIcon",
  // Type tokens
  backgroundButton: "backgroundButton",
  backgroundButtonHover: "backgroundButtonHover",
  backgroundButtonActive: "backgroundButtonActive",
  backgroundButtonFocus: "backgroundButtonFocus",
  colorTextButton: "colorTextButton",
  colorTextButtonHover: "colorTextButtonHover",
  colorTextButtonActive: "colorTextButtonActive",
  borderColorButtonFocus: "borderColorButtonFocus"
};
