// @kiwicom/orbit-components/es/Button/index.js
import _extends from "@babel/runtime/helpers/esm/extends";
import _defineProperty from "@babel/runtime/helpers/esm/defineProperty";
import _objectWithoutProperties from "@babel/runtime/helpers/esm/objectWithoutProperties";

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { _defineProperty(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

import * as React from "react";
import { TYPE_OPTIONS } from "./consts";
import ButtonPrimitive from "../primitives/ButtonPrimitive";
import getIconContainer from "../primitives/ButtonPrimitive/common/getIconContainer";
import getCommonProps from "../primitives/ButtonPrimitive/common/getCommonProps";
import useTheme from "../hooks/useTheme";
import getButtonStyles from "./helpers/getButtonStyles";
import getButtonIconForeground from "./helpers/getButtonIconForeground";
var Button = /*#__PURE__*/React.forwardRef(function (_ref, ref) {
  var _ref$type = _ref.type,
      type = _ref$type === void 0 ? TYPE_OPTIONS.PRIMARY : _ref$type,
      size = _ref.size,
      _ref$disabled = _ref.disabled,
      disabled = _ref$disabled === void 0 ? false : _ref$disabled,
      props = _objectWithoutProperties(_ref, ["type", "size", "disabled"]);

  var theme = useTheme();

  var propsWithTheme = _objectSpread({
    theme: theme,
    size: size
  }, props);

  var commonProps = getCommonProps(propsWithTheme);
  var buttonStyles = getButtonStyles({
    type: type,
    theme: theme,
    disabled: disabled
  });
  var icons = getIconContainer(_objectSpread(_objectSpread({}, propsWithTheme), {}, {
    iconForeground: getButtonIconForeground({
      type: type,
      theme: theme
    })
  }));
  return /*#__PURE__*/React.createElement(ButtonPrimitive, _extends({
    ref: ref,
    disabled: disabled
  }, props, buttonStyles, commonProps, icons));
});
Button.displayName = "Button";
export default Button;
