// @kiwicom/orbit-components/lib/Timeline/TimelineContext.js
"use strict";

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.useStatuses = exports.useStep = exports.TimelineStatusProvider = exports.TimelineStepContext = exports.TimelineStatusContext = void 0;

var _slicedToArray2 = _interopRequireDefault(require("@babel/runtime/helpers/slicedToArray"));

var React = _interopRequireWildcard(require("react"));

var TimelineStatusContext = /*#__PURE__*/React.createContext({
  types: {},
  setTypes: function setTypes() {}
});
exports.TimelineStatusContext = TimelineStatusContext;
var TimelineStepContext = /*#__PURE__*/React.createContext({
  index: 0,
  last: false
});
exports.TimelineStepContext = TimelineStepContext;

var TimelineStatusProvider = function TimelineStatusProvider(_ref) {
  var children = _ref.children;

  var _React$useState = React.useState({}),
      _React$useState2 = (0, _slicedToArray2.default)(_React$useState, 2),
      types = _React$useState2[0],
      setTypes = _React$useState2[1];

  return /*#__PURE__*/React.createElement(TimelineStatusContext.Provider, {
    value: {
      types: types,
      setTypes: setTypes
    }
  }, children);
};

exports.TimelineStatusProvider = TimelineStatusProvider;

var useStep = function useStep() {
  return React.useContext(TimelineStepContext);
};

exports.useStep = useStep;

var useStatuses = function useStatuses() {
  return React.useContext(TimelineStatusContext);
};

exports.useStatuses = useStatuses;
