// @kiwicom/orbit-components/lib/Seat/components/SeatLegend/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = void 0;

var React = _interopRequireWildcard(require("react"));

var _Stack = _interopRequireDefault(require("../../../Stack"));

var _Text = _interopRequireDefault(require("../../../Text"));

var _useTheme = _interopRequireDefault(require("../../../hooks/useTheme"));

var _helpers = require("../helpers");

var _consts = require("../../consts");

var SeatLegend = function SeatLegend(_ref) {
  var _ref$type = _ref.type,
      type = _ref$type === void 0 ? _consts.TYPES.DEFAULT : _ref$type,
      label = _ref.label,
      dataTest = _ref.dataTest;
  var theme = (0, _useTheme.default)();
  return /*#__PURE__*/React.createElement(_Stack.default, {
    inline: true,
    align: "center",
    spacing: "XSmall"
  }, /*#__PURE__*/React.createElement("svg", {
    width: "16",
    height: "20",
    viewBox: "0 0 16 20",
    fill: "none",
    "data-test": dataTest
  }, /*#__PURE__*/React.createElement("path", {
    d: "M0 3C0 1.34315 1.34315 0 3 0H13C14.6569 0 16 1.34315 16 3V19C16 19.5523 15.5523 20 15 20H1C0.447715 20 0 19.5523 0 19V3Z",
    fill: (0, _helpers.resolveAccentColor)({
      type: type,
      theme: theme
    })
  })), label && /*#__PURE__*/React.createElement(_Text.default, {
    type: "secondary",
    size: "small"
  }, label));
};

var _default = SeatLegend;
exports.default = _default;
