// @kiwicom/orbit-components/lib/Table/TableRow/index.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.default = exports.StyledTableRow = void 0;

var React = _interopRequireWildcard(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var StyledTableRow = (0, _styledComponents.default)(function (_ref) {
  var children = _ref.children,
      className = _ref.className,
      dataTest = _ref.dataTest;
  return /*#__PURE__*/React.createElement("tr", {
    className: className,
    "data-test": dataTest
  }, children);
}).withConfig({
  displayName: "TableRow__StyledTableRow",
  componentId: "nnt9s9-0"
})(["box-sizing:border-box;width:100%;white-space:nowrap;"]);
exports.StyledTableRow = StyledTableRow;

var TableRow = function TableRow(_ref2) {
  var dataTest = _ref2.dataTest,
      children = _ref2.children;
  return /*#__PURE__*/React.createElement(StyledTableRow, {
    dataTest: dataTest
  }, children);
};

var _default = TableRow;
exports.default = _default;
